core/settings.lua

```diff
@@ -111,6 +111,10 @@ function settings:declare (spec)
   if spec.name then
     SU.deprecated("'name' argument of SILE.settings:declare", "'parameter' argument of SILE.settings:declare", "0.10.10", "0.11.0")
   end
+  if self.declarations[spec.parameter] then
+    SU.warn("Refusing to declare previously declared setting: " .. spec.parameter)
```

Comment:
That was my own workaround for now... https://github.com/Omikhleia/resilient.sile/blob/main/resilient/hacks.lua#L19-L21

But isn't a warning going to occur _a lot_ in "normal" situations? (I went for a mere debug log.)